### PR TITLE
adding functions and logic for parsing the photo events

### PR DESCRIPTION
**Pull request by ALiwoto**
Signed-off-by: Aliwoto <aminnimaj@gmail.com>

### CODE REVIEW

File: src/gwbot/include/gwbot/lib/tg_event/photo.h [hunk 1]
```diff
@@ -70,12 +119,88 @@ static __always_inline int parse_event_photo(json_object *jmsg,
...
+		ephoto->date = json_object_get_uint64(res);
+	}
+
+
+        if (unlikely(!json_object_object_get_ex(jmsg, "caption", &res))) {
+                /* `caption` is not mandatory, and since there is 
+                 * no caption, there is no `caption_entities` */
+                ephoto->caption = NULL;
+                ephoto->caption_entity_c = 0u;
+		ephoto->caption_entities = NULL;
```
Comment (ammarfaizi2) [inline src/gwbot/include/gwbot/lib/tg_event/photo.h:167]: Indentation

**Review by ammarfaizi2 — CHANGES_REQUESTED** [review]
I can't merge this, you have memory leaks
```
==838564== 
==838564== FILE DESCRIPTORS: 0 open (0 std) at exit.
==838564== 
==838564== HEAP SUMMARY:
==838564==     in use at exit: 144 bytes in 2 blocks
==838564==   total heap usage: 1,627 allocs, 1,625 frees, 143,839 bytes allocated
==838564== 
==838564== 48 bytes in 1 blocks are definitely lost in loss record 1 of 2
==838564==    at 0x4847A25: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==838564==    by 0x1117EF: parse_tgevi_entities (tg_event.h:71)
==838564==    by 0x1117EF: parse_event_photo (photo.h:190)
==838564==    by 0x1117EF: parse_message (tg_event.c:75)
==838564==    by 0x111B4F: internal_tg_event_load_str (tg_event.c:102)
==838564==    by 0x111CA1: tg_event_load_str (tg_event.c:159)
==838564==    by 0x10D1F4: test_tg_event_photo_001_simple (tg_event_photo.c:24)
==838564==    by 0x10AC4F: exec_test (006_tg_event.c:27)
==838564==    by 0x10AF54: main (006_tg_event.c:122)
==838564== 
==838564== 
==838564== Exit program on first error (--exit-on-first-error=yes)
[Thu Mar 25 23:59:42 2021] Error: TEST FAILED!
[Thu Mar 25 23:59:42 2021] Error: Exit code: 99
[Thu Mar 25 23:59:42 2021] Error: Error from valgrind detected
[Thu Mar 25 23:59:42 2021] Error: Please read the error message from valgrind to diagnose this problem
[Thu Mar 25 23:59:42 2021] Error: Reading valgrind backtrace is not trivial, please be serious!
make[1]: *** [/home/ammarfaizi2/project/now/GNUWeebBot/tests/006_tg_event/Makefile:61: /home/ammarfaizi2/project/now/GNUWeebBot/tests/006_tg_event/006_tg_event.test__exec] Error 99
make: *** [/home/ammarfaizi2/project/now/GNUWeebBot/tests/Makefile:21: test] Error 2
```

**Comment by ammarfaizi2** [comment]
Please fix the memory leak issue, we need to reclaim any unused resource.

Every `calloc` must be `free`-ed.

See `tg_event_destroy_text` in `tg_event.c`

**Comment by ammarfaizi2** [comment]
It's very easy to do it myself, but you won't fix memory leak issue in the future if I fix this myself.

**Comment by ALiwoto** [comment]
@ammarfaizi2 thank you for reminding me!
you are the best teacher I've ever had!

**Review by ammarfaizi2 — APPROVED** [review]

**Comment by ammarfaizi2** [comment]
Good job, thanks. I am merging this.

**Pull request merged**